src/homework2/task7.py

```diff
@@ -0,0 +1,233 @@
+    p_sum = principal
+    years = 0
+
+    while p_sum < desired:
+        profit = p_sum * interest
+        p_sum = p_sum + (profit - (profit * tax))
+        years += 1
+
+    return years
+
+
+"""
+    Task 7-2.
+    IQ Test. (6 kyu)
+
+    Bob is preparing to pass IQ test. The most frequent task in this test
+    is to find out which one of the given numbers differs from the others.
+    Bob observed that one number usually differs from the others in evenness.
+    Help Bob — to check his answers, he needs a program that among the given
+    numbers finds one that is different in evenness, and return a position of
+    this number
+
+    ! Keep in mind that your task is to help Bob solve a real IQ test,
+    which means indexes of the elements start from 1 (not 0)
+
+    Examples:
+    iq_test("2 4 7 8 10") => 3 // Third number is odd, while the rest of the
+    numbers are even
+    iq_test("1 2 1 1") => 2 // Second number is even, while the rest of the
+    numbers are odd
+"""
+
+
+def iq_test(numbers):
+    list_of_numbers = numbers.split(" ")
+    odds = []
+    evens = []
+
+    for num in list_of_numbers:
+        index = list_of_numbers.index(num)
+
+        if int(num) % 2:
+            odds.append(index + 1)
+        else:
+            evens.append(index + 1)
+
+    if len(odds) == 1:
+        return odds[0]
+    else:
+        return evens[0]
+
+
+"""
+    Task 7-3
+    Highest Scoring Word (6 kyu)
+
+    Given a string of words, you need to find the highest scoring word.
+    Each letter of a word scores points according to its position in the
+    alphabet:
+    a = 1, b = 2, c = 3 etc.
+
+    You need to return the highest scoring word as a string.
+    If two words score the same, return the word that appears earliest in the
+    original string.
+    All letters will be lowercase and all inputs will be valid.
+
+    Examples:
+    high('man i need a taxi up to ubud') // 'taxi'
+    high('what time are we climbing up the volcano') // 'volcano'
+    high('take me to semynak') // 'semynak'
+"""
+
+
+def high(x):
+    alp = string.ascii_lowercase
+
+    list_of_words = x.split(" ")
+    score_list = {}
+
+    for word in list_of_words:
+        score = 0
+        for symbol in word:
+            score += alp.find(symbol)
+
+        score_list[word] = score
+
+    return max(score_list, key=score_list.get)
+
+
+"""
+    Task 7-4
+    +1 Array (6 kyu)
+
+    Given an array of integers of any length, return an array that has 1 added
+    to the value represented by the array.
+
+    the array can't be empty
+    only non-negative, single digit integers are allowed
+    Return nil (or your language's equivalent) for invalid inputs.
+
+    Examples
+    For example the array [2, 3, 9] equals 239, adding one would return the
+    array [2, 4, 0].
+    [4, 3, 2, 5] would return [4, 3, 2, 6]
+"""
+
+
+def up_array(arr):
+    if not len(arr):
+        return None
+
+    for i in arr:
+        if i < 0 or i > 9:
```

Comment:
0 < i < 9

Reply:
I've tested it in codewars and in this case operation 0 < i < 9 doesn't work

this example can work:
if 0 <= i <= 9:
    continue
else:
    return None

but it looks better in this case when "if" operation writes without else, that's why I used construction "i < 0 or i > 9"